### DISCUSSION TITLE
Stubgen fails to cast enum class defined in other extensions when they are used as default value of function argument

### DISCUSSION THREAD

**Discussion by metab0t** [wjakob/nanobind#621] [Q&A] [answered]
As shown in the demo repo https://github.com/metab0t/nanobind_stubgen_demo

If I use enum defined in an extension module `A` and use it as the default argument value in another extension module `B`, then stubgen will complains bad cast.

The enum is at https://github.com/metab0t/nanobind_stubgen_demo/blob/835e92156a44838b55df03007ce565662c0403b7/src/enum.hpp#L3-L6

and exported at https://github.com/metab0t/nanobind_stubgen_demo/blob/835e92156a44838b55df03007ce565662c0403b7/src/nanobind_example_ext.cpp#L12-L15

Used as default argument value at https://github.com/metab0t/nanobind_stubgen_demo/blob/835e92156a44838b55df03007ce565662c0403b7/src/nanobind_example_ext2.cpp#L9

The `bad cast` error thrown by stubgen is at https://github.com/metab0t/nanobind_stubgen_demo/actions/runs/9498919441/job/26178666710#step:5:151

Is there some way to import other extension modules when generating stubs?

**Comment by metab0t** [comment]
It will be great to have an additional argument to specify the dependent extension module to import as prelude.

**Reply by wjakob** [reply, marked as answer]
In this case, your extension itself should import the other one. See ``nb::module_::import_("...")``

**Reply by metab0t** [reply]
Thanks. I added `nb::module_::import_("A")` to the binding code of module `B` and it works.
